test/e2e/volume_provisioning.go

```diff
@@ -125,6 +126,56 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 
 			testDynamicProvisioning(c, claim)
 		})
+
+		It("should test that deleting a claim before the volume is provisioned deletes the volume. [Volume]", func() {
+			// This case tests for the regressions of a bug fixed by PR #21268
+			// REGRESSION: Deleting the PVC before the PV is provisioned can result in PV
+			// not being deleted
+
+			// Given the small race window, attempt to produce it iteratively
+			var raceAttempts int = 100
+
+			By("creating a StorageClass")
+			class := newStorageClass()
+			class.Name = ns + "-sc" // pseudo-random Storage Class name
+			class, err := c.Storage().StorageClasses().Create(class)
+			defer c.Storage().StorageClasses().Delete(class.Name, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("Creating and deleting PersistentVolumeClaims for %d iterations", raceAttempts))
+			claim := newClaim(ns, false)
+			claim.Annotations[storageutil.StorageClassAnnotation] = class.Name
+			for i := 0; i < raceAttempts; i++ {
+				tmpClaim, err := c.Core().PersistentVolumeClaims(ns).Create(claim)
```

Comment:
Same comment except that var name is "tmpClaim" rather than "claim". The point I'm trying to make (though I might be wrong) is that you are creating a new object for every iteration in the `for`. Now, it's not a big deal because that object is just a ptr in this case. I was thinking it would be "better" to define `claim` outside the loop and just assign it using `=` in the loop. However, using `:=` in `for` loops is common in k8s code, so that's all I'll say on this.